### DISCUSSION TITLE
shfmt: Unexpected behavior in `case` statements

### DISCUSSION THREAD

**Discussion by DannyBen** [mvdan/sh#950] [Q&A] [answered]
Hello all,

I am the developer of [bashly](https://bashly.dannyb.co/), a bash script CLI generator / framework thingy.
All the scripts that bashly generates are shellcheck compliant, and I also want to make them shfmt compliant.

The generated scripts pass shfmt with almost flying colors, except two format suggestions, that seem wrong to me:

1. shfmt adds a backslash in a new line in some `case` statement cases
2. shfmt indents a comment wrongly (perhaps this is similar to #838 ?)

Here is the simplest test case:

```bash
# test.sh
#!/usr/bin/env bash
action=download

case $action in
  -*)
    ;;

  upload)
    # This upload section is being messed up with a weird backslash
    # Curiously, if this comment will be ABOVE the upload, the backslash
    # issue will not happen.
    echo "download"
    ;;

  # This comment is ok (not moving)
  download | d)
    echo "download"
    ;;

  # This comment should not move
  "")
    echo "download"
    ;;

esac
```

and running this:

```
$ shfmt -d -i 2 -ci test.sh
```

yields this:

```diff
 case $action in
-  -*)
-    ;;
+  -*) ;;

-  upload)
+  \
+    upload)
     # This upload section is being messed up with a weird backslash
     # Curiously, if this comment will be ABOVE the upload, the backslash
     # issue will not happen.
     echo "download"
     ;;

-  # This comment should not move
+    # This comment should not move
   "")
     echo "download"
     ;;
```

where I expected it to not suggest any of the changes it did (except maybe the `*) ;;` which I can accept)

**Comment by mvdan** [comment, marked as answer]
Hello, and thanks for raising this!

Your two bugs seem valid, and I think they are very similar to https://github.com/mvdan/sh/issues/779 and https://github.com/mvdan/sh/issues/917. I'll add your examples there, for the sake of avoiding near-duplicates, and take a look when I can.

**Reply by DannyBen** [reply]
Super. I will mark this as Answered then, and subscribe to the other issues.
Thanks.
